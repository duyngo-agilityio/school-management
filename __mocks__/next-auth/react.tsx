import { createContext } from "react";

export const useSession = () => ({
  data: {
    user: {
      token: "access-token",
      email: "user@gmail.com",
      id: "1",
      name: "User",
    },
    expires: Date.now().toString(),
  },
  status: "authenticated",
  update: async () => null,
});

export type Session = {
  data: {
    user: {
      token: string;
      email: string;
      id: string;
      name: string;
    };
    expires: string;
  };
  status: "loading" | "authenticated" | "unauthenticated";
  update: (data?: Session) => Promise<Session | null>;
};

const INITIAL_SESSION: Session = {
  data: {
    user: {
      token: "access-token",
      email: "user@gmail.com",
      id: "1",
      name: "User",
    },
    expires: Date.now().toString(),
  },
  status: "authenticated",
  update: async () => null,
};

const SessionContext = createContext(INITIAL_SESSION);

export const SessionProvider = ({ children }: React.PropsWithChildren) => {
  <SessionContext.Provider value={INITIAL_SESSION}>
    {children}
  </SessionContext.Provider>;
};
