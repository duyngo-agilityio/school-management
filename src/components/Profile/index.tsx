'use client';

// Components
import {
  Avatar,
  Flex,
  FlexProps,
  Text,
  useStyleConfig,
} from '@chakra-ui/react';

type TCustomFlex = 'xs' | 'sm' | 'md' | 'lg';

interface ProfileProps {
  title: string;
  variant: TCustomFlex;
  src?: string;
  id?: string;
  subTitle?: string;
}

interface CustomFlexProps extends FlexProps {
  variant?: TCustomFlex;
}

const CustomFlex = ({ variant, ...props }: CustomFlexProps) => {
  const styles = useStyleConfig('CustomFlex', { variant });

  return <Flex __css={styles} {...props} />;
};

const Profile = ({ id, title, src, subTitle, variant }: ProfileProps) => {
  const variantText = `profile_${variant}`;
  const ml = variant === 'xs' ? '8px' : '';

  const renderAvatar = id ? (
    <Flex
      flexDir="column"
      justifyContent="center"
      alignItems="center"
      gap="12px"
    >
      <Text variant="tertiary">{id}</Text>
      <Avatar src={src} size={variant} />
    </Flex>
  ) : (
    <Avatar src={src} size={variant} />
  );

  const renderTitle = subTitle ? (
    <Flex flexDir="column" alignItems="center" gap="4px">
      <Text variant={variantText}>{title}</Text>
      <Text variant="subTitle">{subTitle}</Text>
    </Flex>
  ) : (
    <Text variant={variantText} ml={ml}>
      {title}
    </Text>
  );

  return (
    <CustomFlex variant={variant}>
      {renderAvatar}
      {renderTitle}
    </CustomFlex>
  );
};

export default Profile;
