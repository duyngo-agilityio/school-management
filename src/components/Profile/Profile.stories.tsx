import type { Meta, StoryObj } from '@storybook/react';

// Components
import Profile from '.';
import { MOCK_STUDENTS } from '@/mocks';
import { Flex } from '@chakra-ui/react';

const meta: Meta<typeof Profile> = {
  title: 'components/Profile',
  component: Profile,
  tags: ['autodocs'],
  argTypes: {
    title: {
      description: 'The main title text displayed in the profile component.',
    },
    variant: {
      description: 'Determines the size of the profile component',
    },
    src: {
      description: 'The source URL for the avatar image',
    },
    id: {
      description: 'Displayed above the avatar within the profile component',
    },
    subTitle: {
      description:
        'An optional subtitle displayed below the main title in the profile component',
    },
  },
};

export default meta;
type Story = StoryObj<typeof Profile>;

export const Default: Story = {
  args: {
    title: MOCK_STUDENTS[0].fullName,
    variant: 'xs',
    src: MOCK_STUDENTS[0].avatar,
  },
};

export const Medium: Story = {
  decorators: [
    (Story) => (
      <Flex
        flexDirection="column"
        w="313px"
        alignItems="center"
        px="39px"
        mt="45px"
      >
        <Story />
      </Flex>
    ),
  ],
  args: {
    id: MOCK_STUDENTS[1].id,
    title: MOCK_STUDENTS[1].fullName,
    src: MOCK_STUDENTS[1].avatar,
    variant: 'md',
    subTitle: 'Physical Student',
  },
};

export const Large: Story = {
  decorators: [
    (Story) => (
      <Flex flexDirection="column" alignItems="center">
        <Story />
      </Flex>
    ),
  ],
  args: {
    title: MOCK_STUDENTS[1].fullName,
    src: MOCK_STUDENTS[1].avatar,
    variant: 'lg',
    subTitle: 'Physical teacher',
  },
};
