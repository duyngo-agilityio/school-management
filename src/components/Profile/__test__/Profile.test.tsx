import { render } from '@testing-library/react';

// Components
import Profile, { TCustomFlex } from '..';

const ProfileProps = {
  title: 'John',
  variant: 'xs' as TCustomFlex,
  src: '',
  id: '123',
  subTitle: 'Math Teacher',
};

const ProfileComponent = (overrideProps = {}) =>
  render(<Profile {...ProfileProps} {...overrideProps} />);

describe('Profile Component', () => {
  it('should match snapshot for Profile', () => {
    const { container } = ProfileComponent();

    expect(container).toMatchSnapshot();
  });

  it('renders match with props variant is md', () => {
    const { container } = ProfileComponent({ variant: 'md' });

    expect(container).toMatchSnapshot();
  });
});
