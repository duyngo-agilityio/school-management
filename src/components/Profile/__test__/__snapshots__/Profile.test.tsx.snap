// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component renders match with props variant is md 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-s1wtk5"
    >
      <p
        class="chakra-text css-0"
      >
        123
      </p>
      <div
        class="css-79elbk"
        style="width: 180px; height: 180px;"
      >
        <img
          alt="John"
          data-nimg="fill"
          decoding="async"
          src=""
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent; border-radius: 50%; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
        />
      </div>
    </div>
    <div
      class="css-izuqqr"
    >
      <p
        class="chakra-text css-0"
      >
        John
      </p>
      <p
        class="chakra-text css-0"
      >
        Math Teacher
      </p>
    </div>
  </div>
</div>
`;

exports[`Profile Component renders match without props id 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-79elbk"
      style="width: 24px; height: 24px;"
    >
      <img
        alt="John"
        data-nimg="fill"
        decoding="async"
        src=""
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent; border-radius: 50%; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
      />
    </div>
    <div
      class="css-izuqqr"
    >
      <p
        class="chakra-text css-0"
      >
        John
      </p>
      <p
        class="chakra-text css-0"
      >
        Math Teacher
      </p>
    </div>
  </div>
</div>
`;

exports[`Profile Component renders match without props subtile 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-s1wtk5"
    >
      <p
        class="chakra-text css-0"
      >
        123
      </p>
      <div
        class="css-79elbk"
        style="width: 24px; height: 24px;"
      >
        <img
          alt="John"
          data-nimg="fill"
          decoding="async"
          src=""
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent; border-radius: 50%; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
        />
      </div>
    </div>
    <p
      class="chakra-text css-1isemmb"
    >
      John
    </p>
  </div>
</div>
`;

exports[`Profile Component should match snapshot for Profile 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-s1wtk5"
    >
      <p
        class="chakra-text css-0"
      >
        123
      </p>
      <div
        class="css-79elbk"
        style="width: 24px; height: 24px;"
      >
        <img
          alt="John"
          data-nimg="fill"
          decoding="async"
          src=""
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent; border-radius: 50%; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
        />
      </div>
    </div>
    <div
      class="css-izuqqr"
    >
      <p
        class="chakra-text css-0"
      >
        John
      </p>
      <p
        class="chakra-text css-0"
      >
        Math Teacher
      </p>
    </div>
  </div>
</div>
`;
